### PR TITLE
Add heroku deploy button

### LINKED ISSUE #5: Add deploy to heroku button
See: https://devcenter.heroku.com/articles/heroku-button

A button should be added to the README that will allow the application to be deployed directly to heroku.

For now, I think the application should be 1:1 to each repository that it monitors, and the repository URL can be configured in the env.

While it might be possible to make the app handle multiple repositories, that will likely require more work as well as some kind of persistence for the application configuration (and auth to allow configuration to authenticated users). I think having it be 1:1 will allow for something working to be created more quickly since we can configure everything through the env.

@mbj wdyt?

### PR DESCRIPTION
**Pull request by dkubb**
This branch will add a button to the README that will allow the application to be deployed to heroku with a single click.
# TODO
- [x] Get instance running on heroku
- [x] Ensure halcyon works for local and heroku compilation
- [x] Confirm all build directories are cached in circleci
- [x] Add Getting Started section to README

[Fix #5]

### CODE REVIEW

**Comment by dkubb** [comment]
Add shell script to setup Heroku environment:

```shell
# TODO: prompt for $app_name
# TODO: prompt for $access_key_id
# TODO: prompt for $secret_access_key
# TODO: prompt for $s3_bucket

heroku git:remote --app $app_name

heroku config:set HALCYON_AWS_ACCESS_KEY_ID=$access_key_id
heroku config:set HALCYON_AWS_SECRET_ACCESS_KEY=$secret_access_key
heroku config:set HALCYON_S3_BUCKET=$s3_bucket

heroku run -s PX build

git commit --amend --no-edit
git push -f heroku master
```

**Comment by dkubb** [comment]
@mbj Do you mind reviewing this?

File: README.md [hunk 1]
```diff
@@ -3,3 +3,27 @@
...
+
+```bash
+# Clone the repository
+git clone https://github.com/dkubb/bot.git
+cd bot
+
+# Setup the local environment
+./sbin/setup.sh
+
+# Build the system locally
```
Comment (mbj) [inline README.md:24]: The build script here follow macosx conventions (installs to `/app`), there is nothing wrong with this, but I think it should be mentioned before asking people to run this script on linux where it'll not work as expected.

Comment (dkubb) [inline README.md:24]: This isn't an OSX convention. I think the halcyon devs opted for this as the default location: https://raw.githubusercontent.com/mietek/halcyon/master/setup.sh

I could look at changing it so it is mounted in the user's home directory if you think it's a good idea.

Comment (mbj) [inline README.md:24]: > I could look at changing it so it is mounted in the user's home directory if you think it's a good idea.

I think it is, `/app` than might be an heroku convention.

Comment (dkubb) [inline README.md:24]: @mbj I couldn't get halcyon to work with anything other than `/app` as the base directory. They must have some of the paths hard-coded in that diretory, or logic that is untested with different paths.

Even still I removed the hard-coded occurrences of `/app` and replaced them with a single configuration variable in the generated `env.sh` file.

**Comment by dkubb** [comment]
@mbj would you mind doing another pass on this?

**Comment by dkubb** [comment]
I am running into lots of problems getting Halycon to work on OSX, so I am going to stop this line of development and possibly look at [stack](https://github.com/commercialhaskell/stack) instead.

Besides I don't really want to deploy to Heroku. I was only looking at it because it _looked_ like it might be simpler than other approaches, but I don't think that's the case with Haskell applications.

**Comment by mbj** [comment]
Agreed.

**Pull request closed without merging**